client/src/components/card-list/card-list.tsx

```diff
@@ -57,18 +66,94 @@ const CardList = (props: CardListProps) => {
 		)
 	})
 
-	return (
-		<div
-			ref={listRef}
-			className={classnames(
-				css.cardList,
-				css[size],
-				wrap === false ? css.noWrap : null
-			)}
-		>
-			{cardsOutput}
-		</div>
+	const stackCards = (cards: Array<CardT>) => {
+		const stack: Map<string, CardT[]> = new Map()
+
+		cards.forEach((card) => {
+			const cardId = card.cardId
+			if (!stack.has(cardId)) {
+				stack.set(cardId, [card])
+			} else {
+				const it = stack.get(cardId)
+				if (!it) return
+				else stack.set(cardId, [...it, card])
+			}
+		})
+		return stack
+	}
+
+	const cardsStackedOutput = Array.from(stackCards(cards)).map(
+		([cardId, cardInstances]) => {
+			if (cardInstances.length === 1) {
+				const info = CARDS[cardId]
+				if (!info) return null
+				const isSelected = equalCard(cardInstances[0], selected)
+				const isPicked = !!picked?.find((pickedCard) =>
+					equalCard(cardInstances[0], pickedCard)
+				)
+				return (
+					<div key={cardId} className={css.card}>
+						<Card
+							onClick={onClick ? () => onClick(cardInstances[0]) : undefined}
+							card={info}
+							selected={isSelected}
+							picked={isPicked}
+						/>
+					</div>
+				)
+			} else {
+				const cardlist = []
+				for (let i = 0; i < cardInstances.length; i++) {
+					const info = CARDS[cardId]
+					if (!info) return null
+					const isSelected = equalCard(cardInstances[i], selected)
+					const isPicked = !!picked?.find((pickedCard) =>
+						equalCard(cardInstances[i], pickedCard)
+					)
+					cardlist.push(
+						<Card
+							onClick={onClick ? () => onClick(cardInstances[i]) : undefined}
+							card={info}
+							selected={isSelected}
+							picked={isPicked}
+						/>
+					)
+				}
+				return (
+					<div key={cardId} className={css.stackcard}>
+						{cardlist}
+					</div>
+				)
+			}
+		}
 	)
+
+	if (stack === true)
```

Comment:
As far as I can see only the output variable differs, why not `{stack ? cardsStackedOutput : cardsOutput}` instead of duplicating the whole thing?